### PR TITLE
[META-56] Add redirect URL to Supabase auth call

### PR DESCRIPTION
**Pull request by cnk1dev**
### Description of issue
Added redirect URL to Supabase auth call so that user is redirected to the correct URL depending on where they come from (dev vs prod environment).

### How I implemented
Added getURL method which determines the correct siteURL and set redirect URL of Supabase auth call to that.

### Other

### CODE REVIEW

File: src/pages/login.tsx [hunk 1]
```diff
@@ -9,6 +9,21 @@ import Loading from './loading'
...
 const inter = Inter({ subsets: ['latin'] })
 
+const getURL = () => {
+  let url =
+    process?.env?.NEXT_PUBLIC_SITE_URL ?? // prod URL
+    process?.env?.NEXT_PUBLIC_VERCEL_URL ?? // automatically set by Vercel
+    'http://localhost:3000/' // dev URL
+
+  // include `https://` when not localhost
+  url = url.includes('http') ? url : `https://${url}`
```
Comment (pczern) [inline src/pages/login.tsx:19]: does Vercel set a url without `https`?

File: src/pages/login.tsx [hunk 2]
```diff
@@ -9,6 +9,21 @@ import Loading from './loading'
 
 const inter = Inter({ subsets: ['latin'] })
 
+const getURL = () => {
+  let url =
+    process?.env?.NEXT_PUBLIC_SITE_URL ?? // prod URL
+    process?.env?.NEXT_PUBLIC_VERCEL_URL ?? // automatically set by Vercel
+    'http://localhost:3000/' // dev URL
```
Comment (pczern) [inline src/pages/login.tsx:16]: is it 3002 or 3000? was there not a change in another PR?

File: src/pages/login.tsx [hunk 3]
```diff
@@ -9,6 +9,21 @@ import Loading from './loading'
 
 const inter = Inter({ subsets: ['latin'] })
 
+const getURL = () => {
```
Comment (pczern) [inline src/pages/login.tsx:12]: do an immediate function invocation and assign the url to a variable

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/pages/login.tsx (hunk 1 above)
Comment (cnk1dev) [reply src/pages/login.tsx:19]: no, why?

File: src/pages/login.tsx (hunk 2 above)
Comment (cnk1dev) [reply src/pages/login.tsx:16]: currently still 3000, PR #31 will set it to 3002 but I'll update it there

File: src/pages/login.tsx (hunk 3 above)
Comment (cnk1dev) [reply src/pages/login.tsx:12]: done

**Review by DogukanGun — APPROVED** [review]

File: src/pages/login.tsx (hunk 1 above)
Comment (pczern) [reply src/pages/login.tsx:19]: it seems like there is no case where the url will be without http prefix

Comment (cnk1dev) [reply src/pages/login.tsx:19]: `NEXT_PUBLIC_SITE_URL` has to be set manually in Vercel, so potentially there. It's just a safeguard though, I took it directly from the docs, I don't see a reason not to use it

Comment (pczern) [reply src/pages/login.tsx:19]: ok then

**Review by pczern — APPROVED** [review]

**Pull request merged**